src/main/java/com/softserve/academy/spaced/repetition/service/CourseService.java

```diff
@@ -151,12 +152,14 @@ public void deleteLocalCourse(Long course_id) throws NotAuthorisedUserException
                 break;
             }
         }
-
         userRepository.save(user);
     }
 
-    public void addDeckToCourse(Long course_id, Long deck_id) {
+    public void addDeckToCourse(Long course_id, Long deck_id) throws SameDeckInCourseException {
         Course course = courseRepository.findOne(course_id);
+        if(course.getDecks().stream().anyMatch(deck -> deck.getId()==deck_id)){
```

Comment:
- As  both deck.getId() and deck_id are both objects, not primitives, equal() should be used.

- Please rename variables according to https://sbforge.org/sonar/rules/show/squid:S00117

Reply:
deck.getId() return long....is that an object?

Reply:
Sorry...it return Long))

Reply:
Ok. Just check this code:

Long v = null;
boolean b = 1L == v;